service/src/main/java/com/kodemon/service/facade/FightFacadeImpl.java

```diff
@@ -0,0 +1,125 @@
+package com.kodemon.service.facade;
+
+import com.kodemon.api.dto.FightDTO;
+import com.kodemon.api.dto.GymDTO;
+import com.kodemon.api.dto.UserDTO;
+import com.kodemon.api.enums.WildPokemonFightMode;
+import com.kodemon.api.facade.FightFacade;
+import com.kodemon.persistence.entity.Gym;
+import com.kodemon.persistence.entity.Pokemon;
+import com.kodemon.persistence.entity.Trainer;
+import com.kodemon.persistence.entity.TrainerFight;
+import com.kodemon.service.interfaces.*;
+import com.kodemon.service.util.Pair;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+
+/**
+ * @author Oliver Roch
+ */
+
+@Service
+@Transactional
+public class FightFacadeImpl implements FightFacade {
+    @Autowired
+    private BeanMappingService beanMappingService;
```

Comment:
Use `@Inject` annotation instead of `@Autowired`. They are both correct, but `@Autowired` is platform (Spring) specific so when we change the DI framework, this will not build.